brickflow/engine/task.py

```diff
@@ -375,6 +388,203 @@ class NotebookTask(JobsTasksNotebookTask):
+        host (str, optional): The host on which the job is to be run. If not provided, a default host is used.
+        token (str, optional): The token used for authentication. If not provided, a default token is used.
+        job_parameters (Any, optional): The parameters for the job. If not provided, default parameters are used.
+
+    Examples:
+        Below are the different ways in which the RunJobTask class can be used inside workflow (run_job_task).:
+            1. RunJobTask(job_name="dev_object_raw_to_cleansed", host="https://nike.cloud.databricks.com",
+            token="***********",job_parameters={"param1": "value1", "param2": "value2"})
+
+            2. RunJobTask(job_name="dev_object_raw_to_cleansed",
+            job_parameters={"param1": "value1", "param2": "value2"})
+
+            3. RunJobTask(job_name="dev_object_raw_to_cleansed")
+    """
+
+    job_name: str
+    job_id: Optional[float] = None
+    host: Optional[str] = None
+    token: Optional[str] = None
+    job_parameters: Optional[Any] = None
+
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+        self.job_id = get_job_id(self.job_name, self.host, self.token)
+        self.job_parameters = kwargs.get("job_parameters", None)
+
+
+class SqlTask(JobsTasksSqlTask):
+    """
+    The SqlTask class is designed to handle SQL tasks in a more specific context.
+
+    An instance of SqlTask represents a SQL task that can be associated with either a SQL query, a file path, an alert,
+    or a dashboard. Each of these associations is represented by a unique identifier (ID).
+    The SqlTask class ensures that only one of these associations is present for any given instance.
+
+    The SqlTask class also provides additional functionality for handling subscriptions. Subscriptions can be paused,
+    and custom subjects can be set for dashboard-related tasks.
+
+    Attributes:
+        dashboard_id (str, optional): The ID of the dashboard.
+        dashboard_custom_subject (str, optional): The custom subject of the dashboard.
+        file_path (str, optional): The path of the SQL file.
+        alert_id (str, optional): The ID of the alert.
+        pause_subscriptions (bool, optional): If true, the alert notifications are not sent to subscribers.
+        subscriptions (dict, optional): The subscriptions for the alert or dashboard.
+        parameters (dict, optional): The parameters for the SQL task.
+        query_id (str, optional): The ID of the SQL query.
+        warehouse_id (str): The ID of the warehouse.
+
+    Examples:
+        Below are the different ways in which the SqlTask class can be used inside workflow (sql_task).:
+            For SQL query: SqlTask(query_id= "4e16dc24-e30d-4683-96d7-cf7da4e263ad", warehouse_id="044a6f42ad7d914a")
+            For SQL file: SqlTask(file_path="/Workspace/Users/raju.gujjalapati@nike.com/raju_sql_test_brickflow.sql",
+              warehouse_id="044a6f42ad7d914a")
```

Comment:
Please remove all the real email id's and id's in the examples

Reply:
Okay